modules/reference/pages/cluster-properties.adoc

```diff
@@ -293,6 +293,46 @@ Max number of simultaneous connections to S3 per shard. Includes connections use
 
 ---
 
+=== cloud_storage_recovery_topic_validation_mode
+
+Validation mode performed before recovering a topic from cloud storage. This property accept the following parameters:
+
+- `no_check`. Skips the checks for topic recovery.
+- `check_manifest_existence`.  Run an existence check of each `partition_manifest`. Fails if there are connection issues to the cloud storage.
+- `check_manifest_and_segment_metadata`. Download the manifest and run a consistency check, comparing the metadata with the cloud storage objects. The process will fail if any metadata references cloud storage objects that are missing.
+
+*Requires restart:* No
+
+*Required:* Yes
+
+*Visibility:* `tunable`
+
+*Type:* string
+
+*Default:* `check_manifest_existence`
+
+*Accepted values:* [`no_check`, `check_manifest_existence`, `check_manifest_and_segment_metadata`]
+
+---
+
+=== cloud_storage_recovery_topic_validation_depth
+
+Number of segment metadata to validate, from newest to oldest, when <<cloud_storage_recovery_topic_validation_mode>> is set to `check_manifest_and_segment_metadata`.
```

Comment:
```suggestion
Number of metadata segments to validate, from newest to oldest, when <<cloud_storage_recovery_topic_validation_mode>> is set to `check_manifest_and_segment_metadata`.
```

@andijcr is this accurate?

Reply:
I think so. The check does not actually read segments, it will  check that each entry in the partition_manifest points to an existing segment in cloud storage, and that other metadata in the entry is consistent